### DISCUSSION TITLE
[tts] 关于synthesize_e2e目录结构的讨论

### DISCUSSION THREAD

**Discussion by jerryuhoo** [PaddlePaddle/PaddleSpeech#1331] [General]
之前我使用synthesize_e2e以及inference的时候会感到有点困惑，因为之前同样在在每个声学模型的目录中也会有一个synthesize_e2e（包括现在的speedyspeech），而在exps这个文件夹中也有一个synthesize_e2e（应该是后来加的吧），为了用一个文件来完成多个声学模型/声码器的合成。

现在可能导致的问题是，随着各种模型的增加（比如之后添加的中文tacotron2），这个总synthesize_e2e会越改越复杂，而这个总synthesize_e2e和其他模型的synthesize_e2e里面代码有一部分重合的，但也缺少了一部分的功能（比如style fastspeech2），然后这个style inference同样也在demo中有一个style_syn.py。所以感觉会比较混乱。

我感觉在每一个模型的分文件夹下添加synthesize_e2e，然后让总synthesize_e2e去调用这些分synthesize_e2e会比较好一点吗？而且需要统一一下目录的结构，比如说把style_syn.py移动到fastspeech的目录下，fastspeech目录下现在没有synthesize_e2e而speedypseech有synthesize_e2e，我觉得每个模型还是需要有自己的synthesize_e2e和inference这两个文件。

**Comment by yt605155624** [comment]
每个模型还是原本有自己的 synthesize_e2e 和 inference 这两个文件，之后删除了（其他模型的 synthesize_e2e 除了 speedyspeech 都已经删除了），是因为这部分代码复制粘贴太严重了，代码很雷同
现在用的是统一的 synthesize_e2e 和 inference，是加了 paddlespeech/cli/tts/infer.py 之后统一改的，我现在大部分模型都是统一的流程（统一的文本前端，声学模型和声码器都有归一化操作），所以其实可以用一套代码

style_syn.py 移动到 fastspeech2 的目录下，这个应该可以，当时只是想增加个 demos，没想这么多，而且参数也没有和现在的 synthesize_e2e 统一

之所以 speedyspeech 保留了 synthesize_e2e 和 inference，是我们有个 benchmark 项目调用了，我暂时懒得改那部分代码🤣
